CONTRIBUTING.md

```diff
@@ -41,6 +41,39 @@ cherry-picked commit:
 $ git pull --rebase upstream master
 ```
 
+Any changes that break consensus must be behind a feature gate and must have
+a merged SIMD.
+
+All changes should have unit and integration tests that cover at least 90% of
+added code paths. These tests should run quickly and not be flaky.
+
+All changes should be stress-tested with relevant test cases, if relevant test cases
+are not present, then write them.
+
+All changes should be benchmarked and evidence posted to the PR.
+Microbenchmark results along with mainnet/testnet validator timings or profiles,
+bench-tps or other relevant integration benchmarks. Any code that adds complexity
+should be justified by a comisurate improvement in speed.
+
+All changes should be reviewed by subject matter experts.
+
+All changes should be merged to master branch first and only critical changes should
+be backported to release branches.
+
+Duplicate code should generally be avoided.
+
+Features should be activated on testnet before mainnet in the closest configuration to mainnet as possible
+Relevant metrics need to be monitored and approriate follow-up given after feature activation.
+
+Avoid “hack” or “one-off” solutions, prefer well-architected designs which are not fragile.
+
+Avoid unnecessary unwrap() or panic()
+
+Don’t break semver, if absolutely necessary increment the appropriate versioning.
```

Comment:
I think the general principle today is to not break public API ever, which often slows down development. If we allow bumping versions for individual agave crates it would help (as api breaks could then be made as often as desired through version bump). Is that the change you have intended?

Reply:
Yes we can break it. I think we should prefer not to break it, but if we do we need to adjust the versioning appropriately so we don't break downstream that automatically picks up minor releases for instance.